src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1956,6 +1957,9 @@ void Unit::AttackerStateUpdate (Unit* victim, WeaponAttackType attType, bool ext
     if (attType != BASE_ATTACK && attType != OFF_ATTACK)
         return;                                             // ignore ranged case
 
+    if (!IsControlledByPlayer())
```

Comment:
correction, HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PLAYER_CONTROLLED).

UNIT_STATE_POSSESSED is only set on CHARM_TYPE_POSSESS but UNIT_FLAG_PLAYER_CONTROLLED is set on both CHARM_TYPE_VEHICLE and CHARM_TYPE_POSSESS